addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -31,6 +37,8 @@ def _process_step(self, discuss_channel):
 
         if self.step_type == 'create_lead':
             self._process_step_create_lead(discuss_channel)
+        elif self.step_type == "create_lead_and_forward":
+            return self._process_step_create_lead_and_forward(discuss_channel, posted_message)
```

Comment:
This return might be wrong, `_process_step` is currently built for a single message, why would we lose the message from `_process_step_create_lead` in favor of this one?
Shouldn't we handle returning multiple messages instead?

Reply:
In my opinion, the message from `_process_step_create_lead` will not be lost because `super()._process_step(discuss_channel)` already posts it. In `_process_step_create_lead_and_forward(discuss_channel, posted_message)`, the function first checks if `posted_message` is not `None`. If `posted_message` is not `None`, the function ensures that the same message is not reposted and directly returns the existing `posted_message`, ensuring that the `create lead & forward` step posts the message only once during the process.